src/tests/Microsoft.PowerFx.Core.Tests/TypeSystemTests/FormulaTypeSerializerSnapshotTests.cs

```diff
@@ -0,0 +1,125 @@
+        /// instead include the subpath bin/(Debug|Release).AnyCPU, depending on whether the assembly was
+        /// built in debug or release mode.
+        /// </summary>
+        private static readonly string _baseDirectory = Path.Join(Directory.GetCurrentDirectory(), "TypeSystemTests", "JsonTypeSnapshots");
+
+        private static readonly string _typeSnapshotDirectory = RegenerateSnapshots ?
+            _baseDirectory
+                .Replace(Path.Join("bin", "Debug", "netcoreapp3.1"), string.Empty)
+                .Replace(Path.Join("bin", "Release", "netcoreapp3.1"), string.Empty) :
+            _baseDirectory;
+
+        private void CheckTypeSnapshot(FormulaType type, string testId, JsonSerializerOptions options)
+        {
+            var directory = _typeSnapshotDirectory;
+            var typeSnapshot = Path.Join(_typeSnapshotDirectory, testId + ".json");
+
+            var actual = JsonSerializer.Serialize(type, options);
+
+            if (File.Exists(typeSnapshot))
+            {
+                var expected = File.ReadAllText(typeSnapshot);
+
+                if (RegenerateSnapshots)
+                {
+#pragma warning disable CS0162 // Unreachable code due to a local switch to regenerate baseline files
+                    try
+                    {
+                        TestUtils.AssertJsonEqual(actual, expected);
+                    }
+                    catch (Xunit.Sdk.XunitException)
+                    {
+                        // Only rewrite if the JSON has meaningfully changed
+                        File.WriteAllText(typeSnapshot, actual + "\r\n");
+                    }
+                }
+                else
+                {
+                    TestUtils.AssertJsonEqual(actual, expected);
+                }
+#pragma warning restore CS0162
+            }
+            else
+            {
+                Assert.True(RegenerateSnapshots, "Snapshot regeneration must be explicitly enabled to make new snapshot tests. Target path is: " + typeSnapshot);
+
+                if (!Directory.Exists(directory))
+                {
+                    Directory.CreateDirectory(directory);
+                }
+
+                File.WriteAllText(typeSnapshot, actual + "\r\n");
+            }
+        }
+
+        /// <summary>
+        /// The expected value of each test is stored in ./JsonTypeSnapshots,
+        /// Named by "simple" + testID.
+        /// </summary>
+        [Theory]
+        [InlineData("PrimitiveNumber", "n")]
+        [InlineData("PrimitiveBoolean", "b")]
+        [InlineData("PrimitiveString", "s")]
+        [InlineData("PrimitiveDate", "D")]
+        [InlineData("PrimitiveTime", "T")]
+        [InlineData("PrimitiveDateTime", "d")]
+        [InlineData("PrimitiveHyperlink", "h")]
+        [InlineData("PrimitiveDTNTZ", "Z")]
+        [InlineData("PrimitiveGuid", "g")]
+        [InlineData("PrimitiveUntyped", "O")]
+        [InlineData("PrimitiveBlank", "N")]
```

Comment:
>"PrimitiveBlank", "N")] [](http://example.com/codeflow?start=20&length=23)

What would be the output for Error type?

Reply:
Good point. I had originally had this throwing on unsupported, but returning None is a better option (and brings it in line with the depth limit as well)